internal/common/objects/MegaBike.go

```diff
@@ -103,3 +106,45 @@ func (mb *MegaBike) UpdateOrientation() {
 		mb.orientation += 2
 	}
 }
+
+// get the count of kicked out agents
+func (mb *MegaBike) GetKickedOutCount() int {
+	return mb.kickedOutCount
+}
+
+// set a current leader
+func (mb *MegaBike) SetCurrentLeader(leaderUUID uuid.UUID) {
+	mb.currentLeaderUUID = leaderUUID
+}
+
+// get the current leader
+func (mb *MegaBike) GetCurrentLeader() uuid.UUID {
+	return mb.currentLeaderUUID
+}
+
+func (mb *MegaBike) KickOutAgent() uuid.UUID {
```

Comment:
the logic might change  a bit when making the agents return a list of agents to kickout (potentially empty), but otherwise looks good